src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable max-statements */
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import MenuSelect, { MenuItem } from '@wallet/components/MenuSelect';
+import Icon from 'src/theme/Icon';
+import useSettings from '@settings/hooks/useSettings';
+import { useBlockchainApplicationMeta } from '@blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta';
+import {
+  useApplicationManagement,
+  useCurrentApplication,
+} from '@blockchainApplication/manage/hooks';
+import { DEFAULT_NETWORK } from 'src/const/config';
+import { Client } from 'src/utils/api/client';
+import styles from './NetworkSwitcherDropdown.css';
+import networks from '../../configuration/networks';
+import { useNetworkStatus } from '../../hooks/queries';
+
+function NetworkSwitcherDropdown({
+  noLabel,
+  onLoadApplications = () => {},
```

Comment:
Would be more true if we used `optional chaining` for these function calls instead of calling a bogus function when these functions are not provided.
`onLoadApplications?.({...})`